### PR TITLE
`to_physical_space` and `to_pixel_index` to accept GeometricTransform

### PR DESCRIPTION
**Pull request by effting**
When [PR-2325](https://github.com/delmic/odemis/pull/2325) is merged, the updated `estimate_grid_orientation` and `estimate_grid_orientation_from_img` now return a GeometricTransform that works on image pixel coordinates. It is desirable to be able to convert this returned transform to work on physical coordinates. This pull request adds this functionality to `to_physical_space` and `to_pixel_index` in `odemis.util.transform`.

### CODE REVIEW

File: src/odemis/util/transform.py
Comment (tepals) [inline src/odemis/util/transform.py:305]: If I understood correctly a `GeometricTransform` can be in physical and in pixel space, it just depends on what input it has. Does this code just assume the caller only calls `to_pixel_index` for geometric transforms that had coordinates in physical space as input and vice versa?

File: src/odemis/util/transform.py
Comment (tepals) [inline src/odemis/util/transform.py:245]: I previously liked that these two functions `to_physical_space` and `to_pixel_index` were really clear and simple; the input and output are both a list of coordinates. 

We could also consider having a to_physical_space and a to_pixel_index method on the `GeometricTransform` class.

File: src/odemis/util/transform.py
Comment (effting) [reply src/odemis/util/transform.py:305]: Typically a `GeometricTransform` is created using its `.from_poinset(src, dst)` method. Consider the following example:
```
src_ji = ...
dst_ji = ...
src_xy = to_physical_space(src_ji, shape, pixel_size)
dst_xy = to_physical_space(dst_ji, shape, pixel_size)
tform_ji = AffineTransform.from_pointset(src_ji, dst_ji)
tform_xy = AffineTransform.from_pointset(src_xy, dst_xy)
```
Then in general `tform_ji` is not equivalent to `tform_xy`. The modified code allows to convert between these two, and indeed assumes you only call `to_physical_space` on something in pixel indices.

File: src/odemis/util/transform.py
Comment (effting) [reply src/odemis/util/transform.py:245]: Then I'd rather make it separate functions. Or just not merge this PR for now and see if it is really that necessary

Comment (pieleric) [reply src/odemis/util/transform.py:245]: I agree with Thera that the original simple version of the function is nicer to read. Object-oriented pattern would indeed suggest to put such function as a method on the class.
That said, it might not be super easy due to (cycle-)dependencies. I'd say it's also fine to create new functions named something like `tform_to_physical_space` and `tform_to_pixel_index`, for just this purpose.

**Review by pieleric — CHANGES_REQUESTED** [review]

File: src/odemis/util/transform.py
Comment (tepals) [reply src/odemis/util/transform.py:245]: Érics idea of having two functions like `tform_to_physical_space` and `tform_to_pixel_index`, sounds like a good idea to me.

**Comment by effting** [comment]
Decision is to not implement this in the current PR. In case of new implementation, a new PR will be made.

**Pull request closed without merging**
